### ISSUE TITLE
TypeError: Cannot read property 'ns' of undefined

### ISSUE THREAD

**Issue by kdaniyar** [gribnoysup/react-yandex-maps#174]
Hi! I'm getting this error with version 4 but version 2.2.0 works ok. I use different machines (windows 10) and try with create-react-app and gatsby but still get the same error. Though same code works in codesandbox.io, with same dependencies(CRA, react-yandex-maps v4). Upgrading node, npm didn't help. Please help, thanks!

````
TypeError: Cannot read property 'ns' of undefined
new t
C:/Users/user/Desktop/my-app/node_modules/react-yandex-maps/dist/production/react-yandex-maps.esm.js:405
  402 | 
  403 | var E = function t(e) {
  404 |   var n = Date.now().toString(32);
> 405 |   this.options = e, this.namespace = e.query.ns || t.defaultQuery.ns, this.onload = t.onloadCallback + "$$" + n, this.onerror = t.onerrorCallback + "$$" + n;
  406 | };
  407 | 
  408 | E.prototype.getApi = function () {
```

**Comment by devasidmi** [comment]
+

**Comment by mmarkelov** [comment]
Your code example will be helpful. Cause now I haven’t got any ideas, what can be wrong

**Comment by kdaniyar** [comment]
1. npx create-react-app my-app
2. npm install --save react-yandex-maps
3: copy paste to app.js:
`
import React from 'react';
import { YMaps } from 'react-yandex-maps';

const App = () => (
  <YMaps>
    <div>My awesome application with maps!</div>
  </YMaps>
);

export default App;
`

And that is it! (on different PCs)

**Comment by mmarkelov** [comment]
@kdaniyar seems like it ok on codesandbox
[![Edit wizardly-sunset-rvpg1](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/wizardly-sunset-rvpg1?fontsize=14)
I'll check it with CRA, but only tomorrow. Please, take a look to provided sandbox, maybe it'll help. Thanks!

**Comment by gribnoysup** [comment]
I can reporoduce it locally with CRA, but yeah, the codesandboxes and the docs are working fine. I guess the webpack configs are different, so sandbox pulls the commonjs one when building and local CRA uses esm. I think the bug might be somehow related to how microbundle produces the `esm` build, but I'll need more time to dig into this. I really don't want to switch back to custom rollout config and would prefer to use microbundle instead...

**Comment by mmarkelov** [comment]
@kdaniyar @devasidmi youp seems like something wrong with build process. Maybe bug with **microbundle**. You can use imports like so, for a while.
```javascript
import React from 'react';
import { YMaps, Map } from "react-yandex-maps/dist/development/react-yandex-maps.js";
...
```

**Comment by gribnoysup** [comment]
`v3` should also work fine, you don't need to downgrade all the way to `v2` for this library to work. Main changes between 2 and 3 were docs and the build process

**Comment by mmarkelov** [comment]
@gribnoysup youp, codesandox is using /dist/development/react-yandex-maps.js and CRA is using esm build.
For esm **YMaps** looks strange - looks like this is not current class:
```
{Children: {…}, createRef: ƒ, Component: ƒ, PureComponent: ƒ, createContext: ƒ, …}
```

**Comment by mmarkelov** [comment]
@gribnoysup just try to use **next** of **microbundle** but no luck here.  Another kind of error :(

**Comment by kdaniyar** [comment]
> @kdaniyar @devasidmi youp seems like something wrong with build process. Maybe bug with **microbundle**. You can use imports like so, for a while.
> 
> ```js
> import React from 'react';
> import { YMaps, Map } from "react-yandex-maps/dist/development/react-yandex-maps.js";
> ...
> ```

Thank's a lot. That worked!

**Comment by gribnoysup** [comment]
I think the root cause can be in this transform:

```js
var C = (function(e) {
  function n(t) {
    e.call(this, t), (this.ymaps = new E(t));
  }
  return (
    e && (n.__proto__ = e),
    ((n.prototype = Object.create(e && e.prototype)).constructor = n),
    (n.prototype.componentDidMount = function() {
      !0 === this.props.preload && this.ymaps.load();
    }),
    (n.prototype.render = function() {
      return t.createElement(
        b.Provider,
        { value: this.ymaps },
        this.props.children
      );
    }),
    n
  );
})(t.Component);
C.defaultProps = {
  version: '2.1',
  enterprise: !1,
  query: { lang: 'ru_RU', load: '', ns: '' },
  preload: !1,
};
```

This is how Provider looks after going through microbundle, (or to be more exact babel if it's used internally). If i'm reading this correct, `defaultProps` are never attached to the minified `Provider` class, so `YMaps` fails trying to get query key from an object

**Comment by mmarkelov** [comment]
@gribnoysup This is because of:
```javascript
var E = function t(e) {
	var n = Date.now().toString(32);
        // t reference to React, but we need YMaps, so it should be E instead of t
	this.options = e, this.namespace = e.query.ns || t.defaultQuery.ns, this.onload = t.onloadCallback + "$$" + n, this.onerror = t.onerrorCallback + "$$" + n;
};
```

**Comment by gribnoysup** [comment]
Should be fixed in `v4.0.1`

**Comment by mmarkelov** [comment]
@gribnoysup yeah! I just thought that using constants will be the rescue )
